packages/kbn-esql-validation-autocomplete/src/autocomplete/autocomplete.test.ts

```diff
@@ -1358,4 +1358,91 @@ describe('autocomplete', () => {
       ).map((s) => (s.text.toLowerCase().includes('null') ? s : attachTriggerCommand(s)))
     );
   });
+
+  describe('Replacement ranges are attached when needed', () => {
+    testSuggestions('FROM a | WHERE doubleField IS NOT N/', [
+      { text: 'IS NOT NULL', rangeToReplace: { start: 28, end: 35 } },
+      { text: 'IS NULL', rangeToReplace: { start: 35, end: 35 } },
+      '!= $0',
+      '< $0',
+      '<= $0',
+      '== $0',
+      '> $0',
+      '>= $0',
+      'IN $0',
+    ]);
+    testSuggestions('FROM a | WHERE doubleField IS N/', [
+      { text: 'IS NOT NULL', rangeToReplace: { start: 28, end: 31 } },
+      { text: 'IS NULL', rangeToReplace: { start: 28, end: 31 } },
+      { text: '!= $0', rangeToReplace: { start: 31, end: 31 } },
```

Comment:
I'm not seeing this behavior in the editor with `from kibana_sample_data_flights  | where AvgTicketPrice is n`. Makes sense that it suggests `IS NOT NULL` and `IS NULL` where the cursor is but why the rest?

Reply:
I had a choice: make it so that the range only gets added for `IS NOT NULL` and `IS NULL` or just add it in a generic way for all.

I opted for the latter because
- including an empty range is valid and does no harm
- it saves us adding a special case
  - if new operators with spaces are ever added, it will work automatically

Reply:
That makes sense. Thanks for the context!